adapter/pkg/operator/utils/utils.go

```diff
@@ -269,3 +276,107 @@ func GetSecretValue(ctx context.Context, client client.Client,
 	}
 	return string(secret.Data[key]), nil
 }
+
+// ResolveAndAddBackendToMapping resolves backend from reference and adds it to the backendMapping.
+func ResolveAndAddBackendToMapping(ctx context.Context, client k8client.Client,
+	backendMapping dpv1alpha1.BackendMapping,
+	backendRef dpv1alpha1.BackendReference, apiPolicyNamespace string) {
+	namespace := gwapiv1b1.Namespace(backendRef.Namespace)
+	backendName := types.NamespacedName{
+		Name:      backendRef.Name,
+		Namespace: GetNamespace(&namespace, apiPolicyNamespace),
+	}
+	backend := GetResolvedBackend(ctx, client, backendName)
```

Comment:
this could be nil right?, do we handle it properly?

Reply:
yes, it is handled, it can be nil until Backend resource is added